kubernetes/resource_kubernetes_tokenrequest.go

```diff
@@ -0,0 +1,177 @@
+package kubernetes
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceKubernetesTokenRequest() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceKubernetesTokenRequestCreate,
+		ReadContext:   resourceKubernetesTokenRequestRead,
+		UpdateContext: resourceKubernetesTokenRequestUpdate,
+		DeleteContext: resourceKubernetesTokenDelete,
+
+		Schema: map[string]*schema.Schema{
+			"api_version": {
+				Type:        schema.TypeString,
+				Description: "The apiVersion of the resource to annotate.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"kind": {
+				Type:        schema.TypeString,
+				Description: "The kind of the resource to annotate.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"metadata": namespacedMetadataSchema("token request", true),
+			"spec": {
+				Type:        schema.TypeList,
+				Description: "Spec of TokenRequest API",
+				Required:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: tokenRequestSpecFields(),
+				},
+			},
+			"status": {
```

Comment:
Exposing Status in a resource isn't advised, as this changes asynchronously on the API side and will cause Terraform to show diffs and changes required to the resource.

A better choice for exposing Status is a datasource, e.g. `kubernetes_token_request_status`